prov/rxd/src/rxd_av.c

```diff
@@ -129,19 +130,14 @@ static fi_addr_t rxd_set_rxd_addr(struct rxd_av *av, fi_addr_t dg_addr)
 
 static fi_addr_t rxd_set_fi_addr(struct rxd_av *av, fi_addr_t rxd_addr)
 {
-	int tries = 0;
-
-	while (av->fi_addr_table[av->fi_addr_idx] != FI_ADDR_UNSPEC &&
-	       tries < av->util_av.count) {
-		if (++av->fi_addr_idx == av->util_av.count)
-			av->fi_addr_idx = 0;
-		tries++;
-	}
-	assert(av->fi_addr_idx < av->util_av.count && tries < av->util_av.count);
-	av->fi_addr_table[av->fi_addr_idx] = rxd_addr;
-	av->rxd_addr_table[rxd_addr].fi_addr = av->fi_addr_idx;
-
-	return av->fi_addr_idx;
+	int index;
+	uint64_t *addr = calloc(1, sizeof(*addr));
+	*addr = rxd_addr;
+	index = ofi_idx_insert(&(av->fi_addr_idx), addr);
+	if (index < 0)
+		return FI_ADDR_UNSPEC;
+	av->rxd_addr_table[rxd_addr].fi_addr = index - 1;
+	return (index - 1);
```

Comment:
addr should be fi_addr_t, rather than u64.

However, the indexer allocates space to store sizeof(void *) at each location.  We can just store rxd_addr directly with something like:
```
ofi_idx_insert(idx, (void *) (uintptr_t) rxd_addr);
```
On 64-bit builds, which will be nearly 100% of the cases, this works without issues.  On a 32-bit build, if it exists, we'll truncate the address to the lower 32-bits, which is still larger the the 1M peers that we can support.

Reply:
The reason I had to allocate memory for holding rxd_addr into addr is because directly storing the pointer to the argument rxd_addr ( though i didnt cast it as a void* uintptr_t type) resulted in the pointer going out of scope after the function exited, and the lookups were causing segmentation faults.  Would this result in a similar behaviour?

Reply:
Trying to store &rxd_addr will not work.  That'll store some address that references the stack.  Storing rxd_addr directly should work, as that just stores the actual value.  You'll need to adjust the lookup call to match.  E.g.
```
fi_addr_t addr;
addr = (intptr_t) ofi_idx_lookup(...);
```